jupyterhub/auth.py

```diff
@@ -721,7 +721,18 @@ def delete_user(self, user):
         All group-assignment APIs are disabled if this is True.
         """,
     )
+    manage_roles = Bool(
+        False,
+        config=True,
+        help="""Let authenticator manage roles
+
+        If True, Authenticator.authenticate and/or .refresh_user
+        may return a list of roles in the 'roles' field,
+        which will be added to the database.
 
+        All role-assignment APIs are disabled if this is True.
```

Comment:
Probably mention `load_roles` config conflict. There aren't currently any role-assignment APIs, but this will be true. Could be a more general statement that all role management will happen through the Authenticator.

Reply:
Addressed in baf6e03c469afce48d4b8195bb89894ad2950bb9 but may need further tuning as we go.